### PR TITLE
Musark 807

### PR DESCRIPTION
**Pull request by rituvesh**

### CODE REVIEW

File: src/modules/objectsearch/ObjectSearchComponent.js [hunk 1]
```diff
@@ -43,31 +53,31 @@ export const ObjectSearchComponent = (props) =>
...
             >
               <FontAwesome name="search" style={{fontSize: '1.3em'}}/>
             </Button>
           </Form>
           <br />
           <h4>
-            {props.loaded &&
-            (props.data.totalMatches > 0 ?
-                I18n.t('musit.objectsearch.results.title', {count: props.data.totalMatches})
+            {props.store.loaded && props.store.data &&
```
Comment (jarlah) [inline src/modules/objectsearch/ObjectSearchComponent.js]: no need to check if store.data exist. It should always exist.

File: src/modules/objectsearch/ObjectSearchComponent.js [hunk 2]
```diff
@@ -43,31 +53,31 @@ export const ObjectSearchComponent = (props) =>
...
+                I18n.t('musit.objectsearch.results.title', {count: props.store.data.totalMatches})
                 :
                 I18n.t('musit.objectsearch.results.noHit')
             )
             }
           </h4>
-          <Loader loaded={!props.loading}>
-          {props.data.matches.length > 0 &&
+          <Loader loaded={!props.store.loading}>
+          {props.store.data && props.store.data.matches.length > 0 &&
```
Comment (jarlah) [inline src/modules/objectsearch/ObjectSearchComponent.js]: no need to check if store.data exists

File: src/modules/objectsearch/ObjectSearchComponent.js [hunk 3]
```diff
@@ -19,13 +20,22 @@ export function renderParam(id, props, style) {
...
         placeholder={I18n.t(`musit.objectsearch.${id}.placeHolder`)}
-        value={props.params[id] || ''}
+        value={props.store.params[id] || ''}
         onChange={(e) => props.onChangeField(id, e.target.value)}
       />
     </FormGroup>
   );
 }
 
+export function callSearchForObjects(props, page) {
```
Comment (jarlah) [inline src/modules/objectsearch/ObjectSearchComponent.js]: rename to searchForObjects. No need to imply that we are calling the method by adding 'call' to its name.

File: src/modules/objectsearch/ObjectSearchComponent.js [hunk 4]
```diff
@@ -142,10 +152,16 @@ export const ObjectSearchComponent = (props) =>
...
             </Table>
             <PagingToolbar
-              numItems={props.data.totalMatches}
-              currentPage={props.params.currentPage}
-              perPage={props.params.perPage}
-              onClick={(page) => props.searchForObjects(props.params, page, props.appSession.getMuseumId(), props.appSession.getCollectionId())}
+              numItems={props.store.data.totalMatches}
+              currentPage={props.store.params.currentPage}
+              perPage={props.store.params.perPage}
+              onClick={(page) => props.searchForObjects({
```
Comment (jarlah) [inline src/modules/objectsearch/ObjectSearchComponent.js]: call searchForObjects here.

File: src/modules/objectsearch/__tests__/ObjectSearchComponent-test.js [hunk 5]
```diff
@@ -42,21 +42,15 @@ describe('ObjectSearchComponent', () => {
...
-        location={{
-          query: {
-            page: '1'
-          },
-          pathname: '/#/search/objects'
+        store={{
+          loaded: true,
+          data: testData,
+          params: {
+            currentPage: 1,
```
Comment (jarlah) [inline src/modules/objectsearch/__tests__/ObjectSearchComponent-test.js:49]: add to params and run test (to update snapshot):
museumNo: '1234',
subNo: '455',
term: 'Lololol',

File: src/modules/objectsearch/__tests__/__snapshots__/ObjectSearchComponent-test.js.snap [hunk 6]
```diff
@@ -35,7 +35,7 @@ exports[`ObjectSearchComponent should display object 1 1`] = `
               placeholder=" "
               style={Object {}}
               type="text"
-              value="1234" />
+              value="" />
```
Comment (jarlah) [inline src/modules/objectsearch/__tests__/__snapshots__/ObjectSearchComponent-test.js.snap]: fix  test, dont fix snapshot :) see comment above in ObjectSearchComponent-test.js.

**Review by jarlah — CHANGES_REQUESTED** [review]
Some fixes must be done, but otherwise good :) :+1:

File: src/modules/objectsearch/ObjectSearchComponent.js (hunk 3 above)
Comment (rituvesh) [reply src/modules/objectsearch/ObjectSearchComponent.js]: Done!

File: src/modules/objectsearch/ObjectSearchComponent.js (hunk 1 above)
Comment (rituvesh) [reply src/modules/objectsearch/ObjectSearchComponent.js]: Done!

File: src/modules/objectsearch/ObjectSearchComponent.js (hunk 2 above)
Comment (rituvesh) [reply src/modules/objectsearch/ObjectSearchComponent.js]: Done!

File: src/modules/objectsearch/ObjectSearchComponent.js (hunk 4 above)
Comment (rituvesh) [reply src/modules/objectsearch/ObjectSearchComponent.js]: Done!

File: src/modules/objectsearch/__tests__/ObjectSearchComponent-test.js (hunk 5 above)
Comment (rituvesh) [reply src/modules/objectsearch/__tests__/ObjectSearchComponent-test.js:49]: Done!

File: src/modules/objectsearch/__tests__/__snapshots__/ObjectSearchComponent-test.js.snap (hunk 6 above)
Comment (rituvesh) [reply src/modules/objectsearch/__tests__/__snapshots__/ObjectSearchComponent-test.js.snap]: Done!

File: src/modules/objectsearch/__tests__/ObjectSearchComponent-test.js (hunk 5 above)
Comment (rituvesh) [reply src/modules/objectsearch/__tests__/ObjectSearchComponent-test.js:49]: Done!

**Review by jarlah — APPROVED** [review]

**Pull request merged**